src/components/WeightedAverages/WeightedAverages.js

```diff
@@ -0,0 +1,150 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { Card, Elevation } from '@blueprintjs/core'
+
+import NoData from 'ui/NoData'
+import Loading from 'ui/Loading'
+import DataTable from 'ui/DataTable'
+import {
+  SectionHeader,
+  SectionHeaderRow,
+  SectionHeaderItem,
+  SectionHeaderTitle,
+  SectionHeaderItemLabel,
+} from 'ui/SectionHeader'
+import TimeRange from 'ui/TimeRange'
+import RefreshButton from 'ui/RefreshButton'
+import MultiPairSelector from 'ui/MultiPairSelector'
+import ClearFiltersButton from 'ui/ClearFiltersButton'
+import {
+  checkInit,
+  checkFetch,
+  togglePair,
+  clearAllPairs,
+} from 'state/utils'
+import queryConstants from 'state/query/constants'
+
+import { getColumns } from './WeightedAverages.columns'
+
+const TYPE = queryConstants.MENU_WEIGHTED_AVERAGES
+
+class WeightedAverages extends PureComponent {
```

Comment:
can this one be refactored to functional component?

Reply:
The class-based approach used here for several reasons:
- codebase consistency, the main part of the report is written as classes for the moment 
- we use unified handlers/helpers like `checkInit`, `checkFetch` across all the reports created for this approach, no need to rework it from scratch just  for one report
- also, functional ones have some bottlenecks, for example, [handling prevProps ](https://blog.logrocket.com/accessing-previous-props-state-react-hooks/) could be a bit tricky part that requires additional logic, with the current approach we have this just out of the box
- think we will rework all to the functions someday, in any case, but global refactoring is out of the scope of this task